### PR TITLE
Fixes #32752 - Don't allow editing readonly settings

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js [hunk 1]
```diff
@@ -14,7 +14,11 @@ const SettingCellInner = props => {
   });
 
   const field = (
-    <span {...rest} className={cssClasses} onClick={() => onEditClick(setting)}>
+    <span
+      {...rest}
+      className={cssClasses}
+      onClick={() => !setting.readonly && onEditClick(setting)}
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js]: This still adds a redundant onClick in read-only fields.
How about using a conditional prop rendering? 

```js
  const onClickHandler = {
    ...(!setting.readonly && { onClick: () => onEditClick(setting) }),
  };

  const field = (
    <span {...rest} className={cssClasses} {...onClickHandler}>
      {valueToString(setting)}
    </span>
  );
```

Comment (tbrisker) [reply webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js]: Thanks for the suggestion! I actually took it one step further in the second commit and moved the entire edit handling into the cell, lmk what you think about it (i'm guessing tests/lint will fail, but didn't want to invest in fixing them before getting feedback on the general approach)

**Review by amirfefer — COMMENTED** [review]
I think this approach simplifies the code a bit, it doesn't make sense to pass the `onEditClick` between nodes.  calling `useDispatch` in a leaf component doesn't affect performance either.

Can you please just fix the failures?

File: webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js [hunk 2]
```diff
@@ -2,19 +2,36 @@ import React from 'react';
...
-  const { setting, className, onEditClick, ...rest } = props;
+  const { setting, className, ...rest } = props;
 
   const cssClasses = classNames(className, {
     'editable-empty': !setting.value && setting.settingsType !== 'boolean',
     'masked-input': setting.encrypted,
   });
 
+  if (!setting.readonly){
+    const { setModalOpen } = useSettingModal();
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js]: hooks cannot be called under conditions

File: webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js [hunk 3]
```diff
@@ -2,19 +2,36 @@ import React from 'react';
...
+    const dispatch = useDispatch();
+    const onEditClick = async setting => {
+      await dispatch(setSettingEditing(setting));
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js]: It isn't related, but `setSettingEditing` isn't async, I wonder why `onEditClick` needs to be asynchronous.

File: webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js (hunk 2 above)
Comment (tbrisker) [reply webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js]: thanks, fixed

File: webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js (hunk 3 above)
Comment (tbrisker) [reply webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js]: looks like it doesn't, fixed

**Comment by tbrisker** [comment]
Updated, I broke my changes down to multiple commits to make the review easier, please squash on merge.

**Comment by tbrisker** [comment]
JS test failure unrelated, should be fixed by GH-8627

File: webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js [hunk 4]
```diff
@@ -2,19 +2,32 @@ import React from 'react';
...
   });
 
+  const { setModalOpen } = useSettingModal();
+  const dispatch = useDispatch();
+  if (!setting.readonly) {
+    rest.onClick = () => {
+      dispatch(setSettingEditing(setting));
+      setModalOpen();
+    };
+  }
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js]: nit: This page will need a PF4 refactoring some day, so I'd say we need to keep it a little more explicit :)

```js
  const canClick = !setting.readonly;
  const openModal = () => {
    dispatch(setSettingEditing(setting));
    setModalOpen();
  };

  const field = (
    <span
      // when `onClick` is undefined  react extracts it from the DOM.
      onClick={canClick ? openModal : undefined}
      {...rest}
      className={cssClasses}
    >
      {valueToString(setting)}
    </span>
  );
```

**Review by amirfefer — COMMENTED** [review]
Thanks @tbrisker 👍 Tested and works well, just a small nitpick

**Comment by tbrisker** [comment]
@amirfefer updated with explicit onclick listener. Don't forget to squash on merge please!

**Review by amirfefer — APPROVED** [review]
Thanks @tbrisker !

**Pull request merged**